source/MaterialXCore/Node.cpp

```diff
@@ -347,6 +352,77 @@ void GraphElement::flattenSubgraphs(const string& target, NodePredicate filter)
                 }
             }
 
+            // Connect any nodegraph outputs within the graph which point to another 
+            // flatten node within the nodegraph. As it's been flattend the previous
+            // reference is incorrect and needs to be updated.
+            if (sourceSubGraph->getOutputCount())
+            {
+                for (OutputPtr sourceOutput : getOutputs())
+                {
+                    const string& nodeNameString = sourceOutput->getNodeName(); 
+                    const string& outputString = sourceOutput->getOutputString(); 
+
+                    if (nodeNameString != processNode->getName())
+                    {
+                        continue;
+                    }
+
+                    // Look for what the original output pointed to.
+                    OutputPtr sourceSubGraphOutput = outputString.empty() ? sourceSubGraph->getOutputs()[0] : sourceSubGraph->getOutput(outputString);
+                    if (!sourceSubGraphOutput)
+                    {
+                        continue;
+                    }
+
+                    string destName = sourceSubGraphOutput->getNodeName();
+                    if (destName.empty())
+                    {
+                        destName = sourceSubGraphOutput->getNodeGraphString();
+                    }
+                    NodePtr sourceSubNode = sourceSubGraph->getNode(destName);
+                    NodePtr destNode = sourceSubNode ? subNodeMap[sourceSubNode] : nullptr;
+                    if (destNode)
+                    {
+                        destName = destNode->getName();
+                    }
+
+                    // Point original output to this one
+                    sourceOutput->setNodeName(destName);
+                }
+            }
+
+            // If the node was flattened then any downstream references
+            // need to be updated to point to the new root of the flatten node.
+            PortElementVec downstreamPorts = downstreamPortMap[processNode];
```

Comment:
Node flattening requiring downstream node output to be remapped. At least appears required if first node flatten is this one.